src/main/java/seedu/address/model/recipe/Recipe.java

```diff
@@ -0,0 +1,126 @@
+package seedu.address.model.recipe;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Recipe in the recipe book. <br>
+ * Guarantees: details are present and not null, field values are validated and immutable.
+ */
+public class Recipe {
+
+    // Identity fields
+    private final Name name;
+    private final List<Ingredient> ingredients = new ArrayList<>();
+
+    // Data fields
+    private final double portion;
+    private final List<String> directions = new ArrayList<>();
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Constructs a {@code Recipe}. <br>
+     *
+     * Every field must be present a not null, otherwise it throws a NullPointerException.
+     */
+    public Recipe(Name name, List<Ingredient> ingredients, int portion, List<String> directions, Set<Tag> tags) {
```

Comment:
Any reason why portion is of type Integer here but is stored as doubled in Data fields?

Reply:
Nice catch, regarding quantity values, I was doing this before Tut 2 would be changing it by the morning and give time for others to check so I can make all the changes at once.